pandas/io/excel/_base.py

```diff
@@ -541,7 +541,7 @@ class ExcelWriter(metaclass=abc.ABCMeta):
 
     Parameters
     ----------
-    path : str
+    path : str or BinaryIO
```

Comment:
we should rename this to `path_or_buf` like all other IO routines, in theory need a deprecation for this.

Reply:
Are you expect do this in current pull request?

Reply:
no this can be done separately (pls make an issue if there isn't one, or if you can just do a PR)